### PR TITLE
Add MWAA module and example

### LINKED ISSUE #441: [BLUEPRINT] - MWAA Workflow

### PR DESCRIPTION
**Pull request by maiconrocha**
### What does this PR do?

Demonstrates how to use Amazon Managed Workflows for Apache Airflow (MWAA) with Amazon EKS.
This example was originated from the steps provided on MWAA documentation on the link below:
[mwaa-eks-example](https://docs.aws.amazon.com/mwaa/latest/userguide/mwaa-eks-example.html)

Also, I have added a module to provision Amazon Managed Workflows for Apache Airflow (MWAA). 

### Motivation
Close #441

As an AWS Employee with experience with EKS, Terraform and MWAA, I want to provide an example to customers in how to use Amazon Managed Workflows for Apache Airflow (MWAA) with Amazon EKS.

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [X] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### Additional Notes

Test Evidence showing the pod was schedule successfully by MWAA:
kubectl get pods -n mwaa
NAME                                             READY   STATUS      RESTARTS   AGE
mwaa-pod-test.2816b0b5b01047af947768d5cd6b889b   0/1     Completed   0          31s

### CODE REVIEW

File: examples/mwaa/main.tf [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  enable_metrics_server     = true
+  enable_cluster_autoscaler = true
+
+}
+
+#------------------------------------------------------------------------
+# AWS MWAA Module
+#------------------------------------------------------------------------
+
+module "mwaa" {
```
Comment (kcoleman731) [inline examples/mwaa/main.tf:201]: Lets move this to the example dir. We are trying to constrain the surface area of eks-blueprints modules to just EKS specific resources.

File: modules/aws-mwaa/main.tf [hunk 2]
```diff
@@ -0,0 +1,354 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# MWAA Environment
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_mwaa_environment" "mwaa" {
```
Comment (kcoleman731) [inline modules/aws-mwaa/main.tf:4]: @drewmullen we should consider creating this as an I&A module?

File: examples/mwaa/main.tf (hunk 1 above)
Comment (maiconrocha) [reply examples/mwaa/main.tf:201]: Thanks @kcoleman731 I have moved the aws-mwaa module to the example dir on the last commit

File: examples/mwaa/dags/mwaa_pod_example.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
```
Comment (vara-bonthu) [inline examples/mwaa/dags/mwaa_pod_example.py:14]: You can remove this license

File: examples/mwaa/aws-mwaa/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# AWS MWAA module
```
Comment (vara-bonthu) [inline examples/mwaa/aws-mwaa/README.md:1]: Raise this module as a PR to `AWS-IA` as a new module

File: examples/mwaa/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+EKS Cluster details can be extracted from terraform output or from AWS Console to get the name of cluster.
+This following command used to update the `kubeconfig` in your local machine where you run kubectl commands to interact with your EKS Cluster.
+
+#### Step 5: Run `update-kubeconfig` command
+
+`~/.kube/config` file gets updated with cluster details and certificate from the below command
+
+    $ aws eks --region <enter-your-region> update-kubeconfig --name <cluster-name>
+
+#### Step 6: List all the worker nodes by running the command below
```
Comment (vara-bonthu) [inline examples/mwaa/README.md:97]: @maiconrocha I think we need more steps to explain how to trigger a job and verify using MWAA. 

It's also useful to add a doc under and explain how to use this example and the features of MWAA etc.. [terraform-aws-eks-blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints)/[docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs/advanced/)

We can extend this doc to trigger emr on eks jobs later

**Review by vara-bonthu — DISMISSED** [review]
Thanks for much needed PR 👍🏼 Few minor comments.
I haven't tested this myself. Please add the testing evidence to this PR.

File: modules/aws-mwaa/main.tf (hunk 2 above)
Comment (maiconrocha) [reply modules/aws-mwaa/main.tf]: Waiting PR to be reviewed: https://github.com/aws-ia/terraform-aws-mwaa/pull/1
Once it is approved the module will be created under aws-ia org, then I can update the example to point to the new location of the module.

File: examples/mwaa/aws-mwaa/README.md (hunk 4 above)
Comment (maiconrocha) [reply examples/mwaa/aws-mwaa/README.md:1]: @vara-bonthu  Waiting PR to be reviewed: https://github.com/aws-ia/terraform-aws-mwaa/pull/1
Once it is approved the module will be created under aws-ia org, then I can update the example to point to the new location of the module.

File: examples/mwaa/dags/mwaa_pod_example.py (hunk 3 above)
Comment (maiconrocha) [reply examples/mwaa/dags/mwaa_pod_example.py]: done

File: examples/mwaa/README.md (hunk 5 above)
Comment (maiconrocha) [reply examples/mwaa/README.md:97]: I have added steps 8,9 and 10 on README to explain how to login to Airflow UI, trigger the DAG and check if the pod was scheduled successfully.

**Comment by vara-bonthu** [comment]
@maiconrocha Please update this PR to point to new [MWAA](https://github.com/aws-ia/terraform-aws-mwaa) module

File: examples/mwaa/main.tf [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+  enable_cluster_autoscaler = false
+
+}
+
+#------------------------------------------------------------------------
+# AWS MWAA Module
+#------------------------------------------------------------------------
+
+module "mwaa" {
+  source                        = "aws-ia/mwaa/aws"
```
Comment (maiconrocha) [inline examples/mwaa/main.tf:202]: @vara-bonthu I'm pointing to aws-ia/mwaa/aws but we haven't published the module there yet, should I create a release on the github module instead and point to the release tag?

Comment (vara-bonthu) [reply examples/mwaa/main.tf:202]: We can merge both the MWAA module and this example at the same time.

**Comment by vara-bonthu** [comment]
@maiconrocha Could you please update this PR as discussed?

**Comment by maiconrocha** [comment]
> @maiconrocha Could you please update this PR as discussed?
What other changes I need to make? I'm point already to mwaa module on aws-ia/mwaa/aws terraform registry, once we published the mwaa module there I can do one more final test before we can merge this PR.

**Comment by maiconrocha** [comment]
@vara-bonthu I have update example to use mwaa module on the registry, it is tested and working as expected

**Comment by bryantbiggs** [comment]
@maiconrocha I'm going to make a few updates but we should be able to get this merged today - thank you again for this contribution!

File: examples/mwaa/main.tf [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+
+#------------------------------------------------------------------------
+# Kubernetes Add-on Module
+#------------------------------------------------------------------------
+module "eks_blueprints_kubernetes_addons" {
+  source = "../../modules/kubernetes-addons"
+
+  eks_cluster_id = module.eks_blueprints.eks_cluster_id
+
+  enable_metrics_server     = false
```
Comment (bryantbiggs) [inline examples/mwaa/main.tf:174]: do we want these enabled or should we just skip the addons for this example? I haven't delved into MWAA but seems like we'd want some autoscaling to support

Comment (bryantbiggs) [reply examples/mwaa/main.tf]: for now I'm enabling them but we can always adjust the example in subsequent PRs

**Review by bryantbiggs — APPROVED** [review]
Thanks for this addition @maiconrocha and the module contribution 🎉

**Pull request merged**